designs/2019-suggestions/README.md

```diff
@@ -0,0 +1,44 @@
+- Start Date: 2019-07-04
+- RFC PR: (leave this empty, to be filled in later)
+- Authors: Ilya Volodin, Dan Abramov
+
+# Code Suggestions
+
+## Summary
+
+This RTC proposes introducing a new mechanism for providing users with suggestions. It's very similar to fixes mechanism that already exists in ESLint, but will provide ability to create multiple suggestions per rule, suggestions will not be automatically applied by using `--fix` flag, and will require user interaction. This feature will only be exposed through the ESLint API, and will not be available on the CLI. 
```

Comment:
Yes, I was struggling with this. On one hand doing multipass autofixes would be great, because suggestion can be processed by other rules and inserted in the right format. On the other, suggestions are not supposed to work like autofixes. It's a manual process to initiate suggestion application to the code, and inserting a suggestion should not affect any code outside of the suggestion code itself. Right now I'm leaning towards inserting suggestions as is, without multipass autofixes applied.

Reply:
I think that's fine. If suggestion is applied, user can execute auto-fix after that to bring it to the same coding standard as the rest of the code. I'll add that in.